domain/relation/state/relation.go

```diff
@@ -25,3 +33,101 @@ func NewState(factory database.TxnRunnerFactory, clock clock.Clock, logger logge
 		logger:    logger,
 	}
 }
+
+// GetRelationEndpointUUID retrieves the endpoint UUID of a given relation
+// for a specific application.
+// It queries the database using the provided application ID and relation UUID
+// arguments.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.ApplicationNotFound] is returned if the application
+//     is not found.
+//   - [relationerrors.RelationEndpointNotFound] is returned is the relation
+//     Endpoint is not found.
+//   - [relationerrors.RelationNotFound] is returned is the relation UUID
+//     is not found.
+func (st *State) GetRelationEndpointUUID(ctx context.Context, args relation.GetRelationEndpointUUIDArgs) (
+	corerelation.EndpointUUID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	type relationEndpointUUID struct {
+		UUID string `db:"uuid"`
+	}
+	type relationEndpointArgs struct {
+		AppID        string `db:"application_uuid"`
+		RelationUUID string `db:"relation_uuid"`
+	}
+	dbArgs := relationEndpointArgs{
+		AppID:        args.ApplicationID.String(),
+		RelationUUID: args.RelationUUID.String(),
+	}
+	stmt, err := st.Prepare(`
+SELECT re.uuid AS &relationEndpointUUID.uuid
+FROM relation_endpoint re
+JOIN application_endpoint ae ON re.endpoint_uuid = ae.uuid
+WHERE ae.application_uuid = $relationEndpointArgs.application_uuid
+AND re.relation_uuid = $relationEndpointArgs.relation_uuid
```

Comment:
Consistency in written SQL makes it much easier.
```suggestion
SELECT re.uuid AS &relationEndpointUUID.uuid
FROM    relation_endpoint re
JOIN      application_endpoint ae ON re.endpoint_uuid = ae.uuid
WHERE  ae.application_uuid = $relationEndpointArgs.application_uuid
AND       re.relation_uuid = $relationEndpointArgs.relation_uuid
```